### ISSUE TITLE
icecast2 and Liquidsoap stopped working

### ISSUE THREAD

**Issue by be2clack** [AzuraCast/AzuraCast#1381]
Icecast encountered an error
[2019-03-31 18:06:37] WARN xml/parsing warning: 
[2019-03-31 18:06:37] WARN xml/parsing unknown element "stream-genre" parsing "mount" at line 29
[2019-03-31 18:10:06] WARN xml/parsing warning: 
[2019-03-31 18:10:06] WARN xml/parsing unknown element "stream-genre" parsing "mount" at line 29
[2019-03-31 18:10:07] WARN xml/parsing warning: 
[2019-03-31 18:10:07] WARN xml/parsing unknown element "stream-genre" parsing "mount" at line 29

Liquidsoap encountered an error
2019/03/31 18:10:41 >>> LOG START
init: exception encountered during start phase:
Address already in use in bind()
2019/03/31 18:10:42 >>> LOG START
init: exception encountered during start phase:
Address already in use in bind()

**Comment by SlvrEagle23** [comment]
@be2clack The first items are just warnings and shouldn't stop the station from operating. You may have an older container still sitting around. Rebooting your machine should fix it.

**Comment by ivellios1988** [comment]
I'm encountering the same error on one of my stations. Icecast shows similar error messages and completely refuses to start.

Icecast's log file says this:

[2019-04-28  15:23:25] WARN xml/parsing warning: 
[2019-04-28  15:23:25] WARN xml/parsing skipping element "stream-url" parsing "mount" at line 28
[2019-04-28  15:23:25] WARN xml/parsing warning: 
[2019-04-28  15:23:25] WARN xml/parsing unknown element "stream-genre" parsing "mount" at line 29

And Liquidsoap's log file says this:

2019/04/28 15:25:48 [testy_2_local_1:2] Connection failed: 401, Authentication Required (HTTP/1.0)
2019/04/28 15:25:48 [testy_2_local_1:3] Will try again in 3.00 sec.
2019/04/28 15:25:50 [lang:3] AzuraCast Raw Response: 
2019/04/28 15:25:50 [lang:3] AzuraCast Error: Delaying subsequent requests...
2019/04/28 15:25:52 [testy_2_local_1:3] Connecting mount /radiodefault for source@127.0.0.1...
strange error flushing buffer ... 
strange error flushing buffer ... 
2019/04/28 15:25:52 [testy_2_local_1:2] Connection failed: 401, Authentication Required (HTTP/1.0)
2019/04/28 15:25:52 [testy_2_local_1:3] Will try again in 3.00 sec.
2019/04/28 15:25:55 [lang:3] AzuraCast Raw Response: 
2019/04/28 15:25:55 [lang:3] AzuraCast Error: Delaying subsequent requests...
2019/04/28 15:25:56 [testy_2_local_1:3] Connecting mount /radiodefault for source@127.0.0.1...
strange error flushing buffer ... 
strange error flushing buffer ... 
2019/04/28 15:25:56 [testy_2_local_1:2] Connection failed: 401, Authentication Required (HTTP/1.0)
2019/04/28 15:25:56 [testy_2_local_1:3] Will try again in 3.00 sec.
2019/04/28 15:26:00 [testy_2_local_1:3] Connecting mount /radiodefault for source@127.0.0.1...
strange error flushing buffer ... 
strange error flushing buffer ... 
2019/04/28 15:26:00 [testy_2_local_1:2] Connection failed: 401, Authentication Required (HTTP/1.0)
2019/04/28 15:26:00 [testy_2_local_1:3] Will try again in 3.00 sec.
2019/04/28 15:26:00 [lang:3] AzuraCast Raw Response: 
2019/04/28 15:26:00 [lang:3] AzuraCast Error: Delaying subsequent requests...
2019/04/28 15:26:01 [single_6433:3] Finished with "/usr/local/share/icecast/web/error.mp3".
2019/04/28 15:26:01 [single_6433:3] Prepared "/usr/local/share/icecast/web/error.mp3" (RID 0).
2019/04/28 15:26:04 [testy_2_local_1:3] Connecting mount /radiodefault for source@127.0.0.1...
strange error flushing buffer ... 
strange error flushing buffer ... 
2019/04/28 15:26:04 [testy_2_local_1:2] Connection failed: 401, Authentication Required (HTTP/1.0)
2019/04/28 15:26:04 [testy_2_local_1:3] Will try again in 3.00 sec.
2019/04/28 15:26:05 [lang:3] AzuraCast Raw Response: 
2019/04/28 15:26:05 [lang:3] AzuraCast Error: Delaying subsequent requests...
2019/04/28 15:26:08 [testy_2_local_1:3] Connecting mount /radiodefault for source@127.0.0.1...
strange error flushing buffer ... 
strange error flushing buffer ... 
2019/04/28 15:26:08 [testy_2_local_1:2] Connection failed: 401, Authentication Required (HTTP/1.0)
2019/04/28 15:26:08 [testy_2_local_1:3] Will try again in 3.00 sec.
2019/04/28 15:26:10 [lang:3] AzuraCast Raw Response: 
2019/04/28 15:26:10 [lang:3] AzuraCast Error: Delaying subsequent requests...
2019/04/28 15:26:12 [testy_2_local_1:3] Connecting mount /radiodefault for source@127.0.0.1...
strange error flushing buffer ... 
strange error flushing buffer ... 
2019/04/28 15:26:12 [testy_2_local_1:2] Connection failed: 401, Authentication Required (HTTP/1.0)
2019/04/28 15:26:12 [testy_2_local_1:3] Will try again in 3.00 sec.

**Comment by ivellios1988** [comment]
I keep testing and testing, and I found that this error occurs when two stations try to broadcast on the same port.

In my AzuraCast installation I created a bunch of stations (currently I have 5 test stations running on my server). I didn't provide custom port numbers for broadcasting so AzuraCast selected the ports automatically. This led to a situation when two stations had broadcasting ports set to 8010. Of course, one station that already occupied this port works well, while the other one cannot start broadcasting and keeps showing error messages. The only solution for me was to edit station's settings and manually change the port for broadcasting.

Probably there is some issue with creating new stations - AzuraCast doesn't check which ports are occupied by other stations and which ones are free.
